### PR TITLE
ci: improve workflow by updating action dependencies and reducing time to spin up docker containers

### PR DESCRIPTION
**Pull request by jeffladiray**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -58,14 +58,9 @@ jobs:
         with:
           path: '**/node_modules'
           key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock', 'packages/**/package.json', 'patches/*.patch') }}
-      - name: Download build
```
Comment (jeffladiray) [inline .github/workflows/build.yml:61]: This is not needed during the lint phase

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -89,9 +84,9 @@ jobs:
           name: build
           path: 'packages/'
       - name: Start docker datasource-sql containers
-        run: docker-compose -f ./packages/datasource-sql/docker-compose.yml up -d; sleep 20
```
Comment (jeffladiray) [inline .github/workflows/build.yml:92]: Not sure why we paid for the 2 * 15s. 5s seems good from my tests, but relying on an healthcheck may be a bit cleaner

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -195,8 +190,7 @@ jobs:
     environment:
       name: github-pages
       url: ${{ steps.deployment.outputs.page_url }}
-    needs: [release]
```
Comment (jeffladiray) [inline .github/workflows/build.yml:198]: I know the release phase also waits from the doc-build and doc-link-validation, so this change is mostly here to ensure a semantical dependency between build the doc and publishing it

File: package.json [hunk 4]
```diff
@@ -42,7 +42,7 @@
     "clean": "lerna run clean && rm -rf coverage api-reference/",
     "docs": "yarn build && typedoc",
     "docs:check-links": "find docs -type f -name '*.md' | xargs -L1 markdown-link-check -q",
-    "lint": "lerna exec eslint src test",
+    "lint": "lerna exec --parallel eslint src test",
```
Comment (jeffladiray) [inline package.json:45]: This was initially a test, but it seems to work just fine + allow to parallelize the linting, which is a bit faster on my machine (50s without, 28s with)

**Review by arnaud-moncel — APPROVED** [review]

**Pull request merged**